src/wfs/src/main/java/org/geoserver/wfs/CatalogFeatureTypeCache.java

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2014 OpenPlans - www.openplans.org. All rights reserved.
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+*/
+package org.geoserver.wfs;
+
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geotools.gml2.FeatureTypeCache;
+import org.geotools.util.logging.Logging;
+import org.opengis.feature.type.FeatureType;
+import org.opengis.feature.type.Name;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Custom FeatureTypeCache that looks up directly from GeoServer catalog.
+ * <p>
+ * This cache class is used by XML bindings during parsing to obtain feature type information.
+ * This custom implementation allows for look up on demand, as opposed to pre-seeding the cache.
+ * </p>
+ */
```

Comment:
It was added to avoid full scans of the catalog for wfs calls which used to populate the cache with every feature type before instantiating xml parsers. I don't recall it being related to pre-seeding.